Singleton/LoggerSingleton.cs

```diff
@@ -0,0 +1,59 @@
+﻿using Microsoft.Win32.SafeHandles;
+using System;
+using System.Runtime.InteropServices;
+using System.IO;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Singleton
+{
+    public sealed class LoggerSingleton : IDisposable
+    {
+        private static readonly Lazy<LoggerSingleton> instance = new Lazy<LoggerSingleton>(() => new LoggerSingleton());
+        private int logID = 1;
+        private StreamWriter logger;
+        private bool disposed = false;
+        private readonly SafeHandle handle = new SafeFileHandle(IntPtr.Zero, true);
+
+        private LoggerSingleton() { }
+
+        public void StartLogger()
+        {
+            Logger = new StreamWriter("Logger1.txt");
+        }
+
+        public void WriteInLog(string logMessage)
+        {
+            DateTime now = DateTime.Now;
+            Logger.Write("Log ID: " + LogID + " Date:"  + now + " " + logMessage + "\n");
+        }
+
+        public void FinishLogger()
```

Comment:
To better understand Singleton pattern write an example where you write log in two threads

Reply:
done